include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,128 @@
+     */
+    long get_data_stop_time() override {
+        //return end_date_time_epoch;
+        return LONG_MAX;
+    }
+
+    /**
+     * @brief the duration of one record of this forcing source
+     *
+     * @return The duration of one record of this forcing source
+     */
+    long record_duration() override {
+        return 1;
+    }
+
+    /**
+     * Get the index of the forcing time step that contains the given point in time.
+     *
+     * An @ref std::out_of_range exception should be thrown if the time is not in any time step.
+     *
+     * @param epoch_time The point in time, as a seconds-based epoch time.
+     * @return The index of the forcing time step that contains the given point in time.
+     * @throws std::out_of_range If the given point is not in any time step.
+     */
+    size_t get_ts_index_for_time(const time_t &epoch_time) override {
+        return 0;
+    }
+
+    /**
+     * Get the value of a forcing property for an arbitrary time period, converting units if needed.
+     *
+     * An @ref std::out_of_range exception should be thrown if the data for the time period is not available.
+     *
+     * @param selector Object storing information about the data to be queried
+     * @param m methode to resample data if needed
+     * @throws std::runtime_error as this provider does not provide forcing value/values
+     */
+    double get_value(const CatchmentAggrDataSelector& selector, data_access::ReSampleMethod m) override
+    {
+        throw std::runtime_error("Called get_value function in NullDataProvider");
+    }
+
+    virtual std::vector<double> get_values(const CatchmentAggrDataSelector& selector, data_access::ReSampleMethod m) override
+    {
+        throw std::runtime_error("Called get_values function in NullDataProvider");
+    }
+
+    /**
+     * Get whether a property's per-time-step values are each an aggregate sum over the entire time step.
+     *
+     * Certain properties, like rain fall, are aggregated sums over an entire time step.  Others, such as pressure,
+     * are not such sums and instead something else like an instantaneous reading or an average value.
+     *
+     * It may be the case that forcing data is needed for some discretization different than the forcing time step.
+     * This aspect must be known in such cases to perform the appropriate value interpolation.
+     *
+     * @param name The name of the forcing property for which the current value is desired.
+     * @return Whether the property's value is an aggregate sum.
+     */
+    //TODO this one used in Bmi_Module_Formulation.hpp and Bmi_Multi_Formulation.hpp
+    inline bool is_property_sum_over_time_step(const std::string& name) override {
+        throw std::runtime_error("Got request for variable " + name + " but no such variable is provided by NullForcingProvider." + SOURCE_LOC);
+    }
+
+    const std::vector<std::string> &get_available_variable_names() override {
```

Comment:
Not an issue of this PR, but I'm annoyed that this interface makes it so that the derived class needs to have an instance of `vector<string>` available to return.

Actually, if it were refactored to return `span<const string>`, then that would go away, since returning an empty temporary `span` is fine. I'll put up a PR for that

Reply:
#644

Reply:
We decided in the standup to get this PR reviewed and integrated first, and then I'll fix up the semantic conflict there